scipy/signal/_waveforms.py

```diff
@@ -33,8 +33,9 @@ def sawtooth(t, width=1):
         Width of the rising ramp as a proportion of the total cycle.
         Default is 1, producing a rising ramp, while 0 produces a falling
         ramp.  `width` = 0.5 produces a triangle wave.
-        If an array, causes wave shape to change over time, and must be the
-        same length as t.
+        If an array, causes wave shape to change over time.  Must be
+        broadcastable with t.
+        Where width not in interval [0, 1], output will be NaN.
```

Comment:
A little nitpick:

```suggestion
        If the wave shape should vary over time, an array with different widths can be 
        used. Its shape has to be either the same shape as `t` or be broadcastable 
        with `t` (as in `~numpy.broadcast_arrays`).
        Where `width` is not in the interval of [0, 1], the output will be NaN.
```

Perhaps a simple example illustrating broadcasting would not hurt.

Reply:
`numpy.arctan2` has the wording below, which is succinct and clear.  I also didn't mention the shape of the output in the docstring update.  What do you think of something on these lines?

```
    If ``x1.shape != x2.shape``, they must be broadcastable to a common
    shape (which becomes the shape of the output).
```

Do we need to elaborate on broadcasting?  It's basic numpy knowledge.  (I'm in favour of terser docs...)

Reply:
That's good to. Maybe adding something like "The output will have the shape ``np.broadcast_shapes(t, w)``" would make it more beginner friendly.

PS: My experience with broadcasting is more or less limited to doing linear algebra wit matrices and tensors. I added the comment, since I find using broadcasting like this not too intuitive.